gcloud-java-dns/src/test/java/com/google/gcloud/dns/testing/LocalDnsHelperTest.java

```diff
@@ -0,0 +1,1429 @@
+    assertNull(zone.getId());
+    options.put(DnsRpc.Option.FIELDS, "description");
+    zone = RPC.getZone(ZONE1.getName(), options);
+    assertNull(zone.getCreationTime());
+    assertNull(zone.getName());
+    assertNull(zone.getDnsName());
+    assertNotNull(zone.getDescription());
+    assertNull(zone.getNameServers());
+    assertNull(zone.getNameServerSet());
+    assertNull(zone.getId());
+    options.put(DnsRpc.Option.FIELDS, "nameServers");
+    zone = RPC.getZone(ZONE1.getName(), options);
+    assertNull(zone.getCreationTime());
+    assertNull(zone.getName());
+    assertNull(zone.getDnsName());
+    assertNull(zone.getDescription());
+    assertNotNull(zone.getNameServers());
+    assertNull(zone.getNameServerSet());
+    assertNull(zone.getId());
+    options.put(DnsRpc.Option.FIELDS, "nameServerSet");
+    zone = RPC.getZone(ZONE1.getName(), options);
+    assertNull(zone.getCreationTime());
+    assertNull(zone.getName());
+    assertNull(zone.getDnsName());
+    assertNull(zone.getDescription());
+    assertNull(zone.getNameServers());
+    assertNotNull(zone.getNameServerSet());
+    assertNull(zone.getId());
+    // several combined
+    options.put(DnsRpc.Option.FIELDS, "nameServerSet,description,id,name");
+    zone = RPC.getZone(ZONE1.getName(), options);
+    assertNull(zone.getCreationTime());
+    assertNotNull(zone.getName());
+    assertNull(zone.getDnsName());
+    assertNotNull(zone.getDescription());
+    assertNull(zone.getNameServers());
+    assertNotNull(zone.getNameServerSet());
+    assertNotNull(zone.getId());
+  }
+
+  @Test
+  public void testDeleteZone() {
+    RPC.create(ZONE1, EMPTY_RPC_OPTIONS);
+    assertTrue(RPC.deleteZone(ZONE1.getName()));
+    assertNull(RPC.getZone(ZONE1.getName(), EMPTY_RPC_OPTIONS));
+    // deleting non-existent zone
+    assertFalse(RPC.deleteZone(ZONE1.getName()));
+    assertNull(RPC.getZone(ZONE1.getName(), EMPTY_RPC_OPTIONS));
+    RPC.create(ZONE1, EMPTY_RPC_OPTIONS);
+    RPC.create(ZONE2, EMPTY_RPC_OPTIONS);
+    assertNotNull(RPC.getZone(ZONE1.getName(), EMPTY_RPC_OPTIONS));
+    assertNotNull(RPC.getZone(ZONE2.getName(), EMPTY_RPC_OPTIONS));
+    // delete in reverse order
+    assertTrue(RPC.deleteZone(ZONE1.getName()));
+    assertNull(RPC.getZone(ZONE1.getName(), EMPTY_RPC_OPTIONS));
+    assertNotNull(RPC.getZone(ZONE2.getName(), EMPTY_RPC_OPTIONS));
+    assertTrue(RPC.deleteZone(ZONE2.getName()));
+    assertNull(RPC.getZone(ZONE1.getName(), EMPTY_RPC_OPTIONS));
+    assertNull(RPC.getZone(ZONE2.getName(), EMPTY_RPC_OPTIONS));
+    RPC.create(ZONE1, EMPTY_RPC_OPTIONS);
+    RPC.applyChangeRequest(ZONE1.getName(), CHANGE_KEEP, EMPTY_RPC_OPTIONS);
+    try {
+      RPC.deleteZone(ZONE1.getName());
+      fail();
+    } catch (DnsException ex) {
+      // expected
+      assertEquals(400, ex.code());
+      assertTrue(ex.getMessage().contains("not empty"));
+    }
+  }
+
+  @Test
+  public void testCreateAndApplyChangeWithThreads() {
```

Comment:
Is it possible to refactor some of the code used here and in the following method (the synchronous version)?